### PR TITLE
cmake build the go executables 

### PR DESCRIPTION
**Pull request by MohammadAlTurany**
Add the go example done by Sebastien Binet (@sbinet) to the cmake (pull request #313)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+ #         GNU Lesser General Public Licence version 3 (LGPL) version 3,        #
  #                  copied verbatim in the file "LICENSE"                       #
  ################################################################################
 # The name of our project is "FairBase".  CMakeLists files in this project can
 # refer to the root source directory of the project as ${FAIRROOT_SOURCE_DIR} or
-# as ${CMAKE_SOURCE_DIR} and to the root binary directory of the project as 
+# as ${CMAKE_SOURCE_DIR} and to the root binary directory of the project as
 # ${FAIRROOT_BINARY_DIR} or ${CMAKE_BINARY_DIR}.
-# This difference is is important for the base classes which are in FAIRROOT 
+# This difference is is important for the base classes which are in FAIRROOT
```
Comment (sbinet) [inline CMakeLists.txt]:
```
-This difference is is important
+This difference is important
```

File: examples/advanced/GoTutorial/src/go-processor/README.md [hunk 2]
```diff
@@ -3,7 +3,9 @@
 `go-processor` is a simple `Go` based program using `nanomsg` as a transport facility.
 It replicates the basic example from [FairRootMQ example 2](https://github.com/FairRootGroup/FairRoot/tree/master/examples/MQ/2-sampler-processor-sink).
 
-## Build
+## Manual Build
+Cmake take care of building this example (if Go is available!), the binaries are in the same directory with the source up to now, but the make install will copy them to the install directory with all other excutables, However if you went to build them manually:
```
Comment (sbinet) [inline examples/advanced/GoTutorial/src/go-processor/README.md]:
```
-Cmake take care
+Cmake takes care
```

File: examples/advanced/GoTutorial/src/go-sink/README.md [hunk 3]
```diff
@@ -3,7 +3,8 @@
 `go-sink` is a simple `Go` based program using `nanomsg` as a transport facility.
 It replicates the basic example from [FairRootMQ example 1](https://github.com/FairRootGroup/FairRoot/tree/master/examples/MQ/1-sampler-sink).
 
-## Build
+## Manual Build
+Cmake take care of building this example (if Go is available!), the binaries are in the same directory with the source up to now, but the make install will copy them to the install directory with all other excutables, However if you went to build them manually:
```
Comment (sbinet) [inline examples/advanced/GoTutorial/src/go-sink/README.md]:
```
-Cmake take care
+Cmake takes care
```

File: examples/advanced/GoTutorial/src/go-processor/README.md (hunk 2 above)
Comment (sbinet) [inline examples/advanced/GoTutorial/src/go-processor/README.md]:
```
-all other excutables
+all other executables
```

File: examples/advanced/GoTutorial/src/go-sink/README.md (hunk 3 above)
Comment (sbinet) [inline examples/advanced/GoTutorial/src/go-sink/README.md]:
```
-all other excutables
+all other executables
```

File: examples/advanced/GoTutorial/src/go-processor/README.md (hunk 2 above)
Comment (MohammadAlTurany) [inline examples/advanced/GoTutorial/src/go-processor/README.md]: done! Thanks

File: examples/advanced/GoTutorial/src/go-sink/README.md (hunk 3 above)
Comment (MohammadAlTurany) [inline examples/advanced/GoTutorial/src/go-sink/README.md]: Done! Thanks

File: examples/advanced/GoTutorial/CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
-export GOPATH=`pwd`:$GOPATH
-GOBIN=$CMAKE_INSTALL_PATH/bin go get go-sink
-GOBIN=$CMAKE_INSTALL_PATH/bin go get go-processor
 
+set(GOPATH $ENV{GOPATH})
+set(GOPATH ${CMAKE_SOURCE_DIR}/examples/advanced/GoTutorial/)
+set(ENV{GOPATH} "${GOPATH}")
+
+
+execute_process(COMMAND ${GO_EXECUTABLE} get go-processor WORKING_DIRECTORY ${GOPATH}/src/go-processor)
```
Comment (sbinet) [inline examples/advanced/GoTutorial/CMakeLists.txt]: Running both 'go get' and 'go build' seems suspicious.
Let me check that...

**Pull request merged**